test/features/categories/categories.screen_test.dart

```diff
@@ -69,5 +69,28 @@ void main() {
     // then
     expect(find.text(mockAnimalCategory.name), findsOneWidget);
     expect(find.text(mockTransportCategory.name), findsOneWidget);
+    expect(find.text(mockColorsCategory.name), findsOneWidget);
+    expect(find.text(mockUSStatesCategory.name), findsOneWidget);
+    expect(find.text(mockPlanetsCategory.name), findsOneWidget);
+    expect(find.text(mockCountriesCategory.name), findsOneWidget);
+  });
+
+  testWidgets('Should handle the user choice on press', (WidgetTester tester) async {
```

Comment:
Suggestion: can you also add another test to verify the expected _navigation _ has been triggered?
```dart
testWidgets('Should navigate back on press', ...
   ...
   verify(mockNavigationObserver.didPop(any));
```

💡 [How to test navigation via Navigator in Flutter](https://stackoverflow.com/a/51983194/704681)